src/test/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/integration/JwtAuthHelper.kt

```diff
@@ -0,0 +1,65 @@
+package uk.gov.justice.digital.hmpps.digitalprisonreportingmi.integration
+
+import io.jsonwebtoken.Jwts
+import io.jsonwebtoken.SignatureAlgorithm
+import org.springframework.context.annotation.Bean
+import org.springframework.http.HttpHeaders
+import org.springframework.security.oauth2.jwt.JwtDecoder
+import org.springframework.security.oauth2.jwt.NimbusJwtDecoder
+import org.springframework.stereotype.Component
+import java.security.KeyPair
+import java.security.KeyPairGenerator
+import java.security.interfaces.RSAPublicKey
+import java.time.Duration
+import java.util.Date
+import java.util.UUID
+
+@Component
+class JwtAuthHelper {
+  private val keyPair: KeyPair
+
+  init {
+    val gen = KeyPairGenerator.getInstance("RSA")
+    gen.initialize(2048)
+    keyPair = gen.generateKeyPair()
+  }
+
+  @Bean
+  fun jwtDecoder(): JwtDecoder = NimbusJwtDecoder.withPublicKey(keyPair.public as RSAPublicKey).build()
+
+  fun setAuthorisation(
+    user: String = "prison-reporting-mi-client",
+    roles: List<String> = listOf(),
+    scopes: List<String> = listOf(),
+  ): (HttpHeaders) -> Unit {
+    val token = createJwt(
+      subject = user,
+      scope = scopes,
+      expiryTime = Duration.ofHours(1L),
+      roles = roles,
+    )
+    return { it.set(HttpHeaders.AUTHORIZATION, "Bearer $token") }
+  }
+
+  internal fun createJwt(
+    subject: String?,
+    scope: List<String>? = listOf(),
```

Comment:
Will the scope and roles lists ever be null if you're providing a default value of an empty list?

Reply:
They will not be null as the calling method (setAuthorisation) defaults to empty lists. Making it nullable just allows other callers to pass in null. This was actually copied from the visit scheduler project so ideally it would be best to move it to a library in the future and reuse it. I am happy to change this to non nullable in this case though.

Reply:
Your call, but I think it would clearer to have this non-nullable and it means you can simplify how you build up the hash (as per my second comment).

Reply:
Done